src/schedule.cr

```diff
@@ -56,6 +79,46 @@ module Schedule
     raise RetryException.new
   end
 
+  def self.calculate_interval(interval : Symbol)
+    INTERVALS[interval]
```

Comment:
The way it is implemented now is the same as calling ```Schedule.every(1.hour)```. This method should calculate the interval to the next execution, so a ```Schedule.every :hour``` has to calculate the time between now and the next hour o'clock (01:00, 02:00...).

So you can use [at_the_end_of_*](https://crystal-lang.org/api/0.23.1/Time.html#at_end_of_hour-instance-method) methods.

```crystal
def self.calculate_interval(interval : Symbol)
  now = Time.now
  case interval
  when :minute
    now.at_end_of_minute - now
  when :hour
    now.at_end_of_hour - now
  # ...
  end
end
```

Reply:
Yes. Definitely a better usage of `:hour` symbol, I will implement that.